### PR TITLE
Use prettier 2.x - Root package.json has stale dependencies (prettier 1, etc..) #1326

### PR DESCRIPTION
**Pull request by christophpurrer**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

See https://github.com/microsoft/react-native-macos/issues/1326

- This change only fixes the inconsistent setting of prettier 1.x vs prettier 2.x
- facebook/react-native switched to prettier 2.x in https://github.com/facebook/react-native/commit/f3930a93d5cf5f19058b0bfe453c7b9324eb8552
- package.json still needs further clean up

Right now I can't run ```yarn install``` as I encounter the following error
```
chpurrer@chpurrer-mbp react-native-macos % yarn
yarn install v1.17.0-20190429.1820
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning " > @react-native-community/cli@6.4.0" has unmet peer dependency "react-native@*".
warning "@react-native-community/cli > @react-native-community/cli-plugin-metro > metro-react-native-babel-transformer@0.66.2" has unmet peer dependency "@babel/core@*".
warning " > metro-react-native-babel-transformer@0.67.0" has unmet peer dependency "@babel/core@*".
warning "react-native-codegen > jscodeshift@0.11.0" has unmet peer dependency "@babel/preset-env@^7.1.6".
warning " > react-shallow-renderer@16.14.1" has unmet peer dependency "react@^16.0.0 || ^17.0.0".
warning " > use-subscription@1.8.0" has unmet peer dependency "react@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "use-subscription > use-sync-external-store@1.2.0" has unmet peer dependency "react@^16.8.0 || ^17.0.0 || ^18.0.0".
warning " > workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c@1000.0.0" has unmet peer dependency "react@17.0.2".
warning " > @react-native/tester@0.0.1" has unmet peer dependency "react@17.0.2".
warning " > @react-native/tester@0.0.1" has unmet peer dependency "react-native@*".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > react-native-codegen > jscodeshift@0.13.1" has unmet peer dependency "@babel/preset-env@^7.1.6".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-config-fb-strict@24.9.0" has incorrect peer dependency "eslint@^4.2.0 || ^5.0.0".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-config-fbjs@2.1.0" has incorrect peer dependency "babel-eslint@^7.2.3 || ^8.0.0 || ^9.0.0".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-config-fbjs@2.1.0" has incorrect peer dependency "eslint@^4.2.0 || ^5.0.0".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-config-fbjs@2.1.0" has incorrect peer dependency "eslint-plugin-relay@~0.0.8".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-plugin-jsx-a11y@6.2.1" has incorrect peer dependency "eslint@^3 || ^4 || ^5".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > react-native-macos-init > just-scripts > just-scripts-utils > marked-terminal@4.2.0" has incorrect peer dependency "marked@^1.0.0 || ^2.0.0".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > eslint-plugin-react@7.12.4" has incorrect peer dependency "eslint@^3.0.0 || ^4.0.0 || ^5.0.0".
warning "workspace-aggregator-717c3345-927c-4557-a465-e1be3d29816c > @react-native/repo-config > jscodeshift@0.9.0" has unmet peer dependency "@babel/preset-env@^7.1.6".
error An unexpected error occurred: "could not find a copy of jscodeshift to link in /Users/chpurrer/Documents/GitHub/react-native-macos/node_modules/@react-native/babel-plugin-codegen/node_modules/react-native-codegen/node_modules".
info If you think this is a bug, please open a bug report with the information provided in "/Users/chpurrer/Documents/GitHub/react-native-macos/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
chpurrer@chpurrer-mbp react-native-macos % yarn prettier --fix
yarn run v1.17.0-20190429.1820
$ prettier --write "./**/*.{js,md,yml}" --fix
/bin/sh: prettier: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

With the change here I can run ```yarn install```

## Changelog

[macOS] [Fixed] - Use prettier 2.x - Root package.json has stale dependencies (prettier 1, etc..) #1326

## Test Plan

Ran
```
yarn prettier --fix
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -163,7 +163,6 @@
     "metro-babel-register": "0.66.2",
     "metro-transform-plugins": "^0.66.2",
     "mkdirp": "^0.5.1",
-    "prettier": "1.19.1",
```
Comment (Saadnajmi) [inline package.json:166]: From what I recall, we can actually remove a lot of the dependencies here. Keeping this scoped to prettier is fine, but I'd keep the original bug I filed open so we can remember to remove the rest.

**Review by Saadnajmi — APPROVED** [review]

**Review by mischreiber — APPROVED** [review]

**Comment by christophpurrer** [comment]
@Saadnajmi - some build jobs are failing. E.g Apple PR (JavaScript React Native PR office) - I guess those jobs still use prettier 1.x ....
How can we change this?

**Comment by Saadnajmi** [comment]
> @Saadnajmi - some build jobs are failing. E.g Apple PR (JavaScript React Native PR office) - I guess those jobs still use prettier 1.x ....
> How can we change this?

Huh. Those jobs are defined in `.ado/apple-pr.yml` , but I don't see anything obviously forcing prettier in there. IT feels like line 15 of `.ado/apple-job-javascript` isn't doing a proper yarn install. Unfortuately, Azure pipelines are pretty difficult to debug without just running it once per change :/

**Comment by Saadnajmi** [comment]
@christophpurrer This PR might fix the CI https://github.com/microsoft/react-native-macos/pull/1329

**Comment by christophpurrer** [comment]
> @christophpurrer This PR might fix the CI #1329

ship it :-)

File: package.json (hunk 1 above)
Comment (Saadnajmi) [reply package.json:166]: Actually, RN Core @ 0.68-stable has prettier listed at 2.4.1 here https://github.com/facebook/react-native/blob/4555fc7865b00f598caf86eb0e937bfac6901cc5/package.json#L155

I think you need to add this back, which might also fix CI. Once this merges, I think I'll do a larger cleanup of the package.jsons files at root and repo-config.

Comment (christophpurrer) [reply package.json:166]: I have brought prettier ^2.4.1 - but it did not seem to help with the CI builds ...

**Comment by christophpurrer** [comment]
Abandoning in favor of https://github.com/microsoft/react-native-macos/pull/1333

**Pull request closed without merging**